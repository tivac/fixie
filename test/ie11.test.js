"use strict";

var simple = require("./util.js").simple,
    nested = require("./util.js").nested;

describe(":ie11", () => {
    it("should transform :ie11(...) into an ie11 hack (single selector)", () =>
        simple(`
            :ie11(.fooga) {
                color: red;
            }
        `)
    );

    it("should transform :ie11(...) into an ie11 hack (multiple selectors)", () =>
        simple(`
            :ie11(.fooga .wooga) {
                color: red;
            }
        `)
    );

    it("should work with postcss-nested", () =>
        nested(`
            .fooga {
                color: red;
                
                :ie11(&) {
                    color: blue;
                }
            }
        `)
    );
});
