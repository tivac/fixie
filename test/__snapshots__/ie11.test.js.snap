// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:ie11 should transform :ie11(...) into an ie11 hack (multiple selectors) 1`] = `
"_:-ms-fullscreen,:root .fooga .wooga {
    color: red;
}"
`;

exports[`:ie11 should transform :ie11(...) into an ie11 hack (single selector) 1`] = `
"_:-ms-fullscreen,:root .fooga {
    color: red;
}"
`;

exports[`:ie11 should work with postcss-nested 1`] = `
".fooga {
    color: red;
}
    
    _:-ms-fullscreen,:root .fooga {
        color: blue;
    }"
`;
